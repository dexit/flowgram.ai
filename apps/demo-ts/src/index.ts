export * from './types'
export * from './runtime'
export * from './translate'
export * from './validate'
export * from './util'
