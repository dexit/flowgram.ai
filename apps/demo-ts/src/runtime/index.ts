export * from './run'
export * from './RuntimeError'
