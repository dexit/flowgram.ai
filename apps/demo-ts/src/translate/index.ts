export * from './nodes'
export * from './translate'
