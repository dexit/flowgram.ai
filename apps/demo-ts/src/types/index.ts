export * from '@flow-ide-editor/schema'
export * from './nodes'

export class InvalidSchemaError extends Error {}
export class InvalidNodeTypeError extends Error {}
export class InvalidTypeError extends Error {}
export class NotImplementedError extends Error {}

export type FlowArgs = Record<string, unknown>
