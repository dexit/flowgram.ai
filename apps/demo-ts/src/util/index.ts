export * from './log'
export * from './JSBuffer'
