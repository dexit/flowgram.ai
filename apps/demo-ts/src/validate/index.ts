import { FlowNodeSchema } from '@flow-ide-editor/schema'
import { InvalidSchemaError } from '../types'

// TODO use json-schema validation
export function validateOrThrow(node: FlowNodeSchema): void {
  if (!node.definition) throw new InvalidSchemaError()
  if (!node.definition.info) throw new InvalidSchemaError()
  if (!node.definition.info.title) throw new InvalidSchemaError()
}
