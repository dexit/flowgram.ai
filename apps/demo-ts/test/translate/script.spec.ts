import _ from 'lodash'
import { FlowNodeSchema, translateNode } from '../../src'

const script: FlowNodeSchema = {
  id: 'script_0033f732f85',
  type: 'script',
  definition: {
    info: {
      title: 'script_1',
    },
    inputs: {
      properties: { name: { title: 'Name', type: 'string' } },
      type: 'object',
    },
    outputs: {
      properties: { output: { title: 'Output', type: 'string' } },
      title: 'Output',
      type: 'object',
    },
    script: {
      language: 'python',
      content: 'outputs.output = inputs.name',
    },
  },
  data: {},
}

describe('translate', () => {
  it('Script - empty', () => {
    const node = _.cloneDeep(script)
    node.definition!.script!.content = ''

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Script - basic', () => {
    const node = _.cloneDeep(script)

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Script - complex', () => {
    const node = _.cloneDeep(script)
    node.definition!.script!.content = `function insertionSort(arr) {
for (let i = 1; i < arr.length; i++) {
    let key = arr[i];
    let j = i - 1;

    // Move elements of arr[0..i-1], that are greater than key, to one position ahead of their current position
    while (j >= 0 && arr[j] > key) {
        arr[j + 1] = arr[j];
        j = j - 1;
    }
    arr[j + 1] = key;
}
return arr;
}`

    expect(translateNode(node)).toMatchSnapshot()
  })
})
