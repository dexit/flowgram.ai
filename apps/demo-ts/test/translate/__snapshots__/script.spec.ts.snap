// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate Script - basic 1`] = `
"  //=== script_1@script | {level:1,blockIndex:0}
  const script_1 = {}
  script_1.outputs = {
    output: '',
  }
  script_1.data = {
  }
  const __func_script_1 = (inputs, outputs) => {
  outputs.output = inputs.name
  }
  __func_script_1(script_1.data, script_1.outputs)
  "
`;

exports[`translate Script - complex 1`] = `
"  //=== script_1@script | {level:1,blockIndex:0}
  const script_1 = {}
  script_1.outputs = {
    output: '',
  }
  script_1.data = {
  }
  const __func_script_1 = (inputs, outputs) => {
  function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
      let key = arr[i];
      let j = i - 1;
  
      // Move elements of arr[0..i-1], that are greater than key, to one position ahead of their current position
      while (j >= 0 && arr[j] > key) {
          arr[j + 1] = arr[j];
          j = j - 1;
      }
      arr[j + 1] = key;
  }
  return arr;
  }
  }
  __func_script_1(script_1.data, script_1.outputs)
  "
`;

exports[`translate Script - empty 1`] = `
"  //=== script_1@script | {level:1,blockIndex:0}
  const script_1 = {}
  script_1.outputs = {
    output: '',
  }
  script_1.data = {
  }
  const __func_script_1 = (inputs, outputs) => {
  
  }
  __func_script_1(script_1.data, script_1.outputs)
  "
`;
